src/client/array/dc_array.c

```diff
@@ -208,30 +217,35 @@ create_handle_cb(tse_task_t *task, void *data)
 	if (array == NULL)
 		D_GOTO(err_obj, rc = -DER_NOMEM);
 
-	array->coh = args->coh;
-	array->oid.hi = args->oid.hi;
-	array->oid.lo = args->oid.lo;
-	array->mode = DAOS_OO_RW;
-	array->cell_size = args->cell_size;
-	array->chunk_size = args->chunk_size;
-	array->daos_oh = *args->oh;
+	array->coh		= args->coh;
+	array->oid.hi		= args->oid.hi;
+	array->oid.lo		= args->oid.lo;
+	array->mode		= DAOS_OO_RW;
+	array->cell_size	= args->cell_size;
+	array->chunk_size	= args->chunk_size;
+	array->daos_oh		= *args->oh;
+
+	feat = daos_obj_id2feat(args->oid);
+	if (feat & DAOS_OF_ARRAY_BYTE)
+		array->byte_array = true;
```

Comment:
so is this only an option for arrays that are bytes and not for any other record size?  What if a user has same issue with their custom record size array where they want to zero holes?

Reply:
yes this is only intended for byte arrays / dfs files.
short read handling and hole management is usually only for files. a user working with arrays (single or multi dimensional) does not usually expect this behavior.